### ISSUE TITLE
react_component helper not rendering component

### ISSUE THREAD

**Issue by chrismv48** [reactjs/react-rails#725]
Help us help you! Please choose one:

- [ ] My app crashes with `react-rails`, so I've included the stack trace and the exact steps which make it crash.
- [x] My app doesn't crash, but I'm getting unexpected behavior. So, I've described the unexpected behavior and suggested a new behavior.
- [ ] I'm trying to use `react-rails` with another library, but I'm having trouble. I've described my JavaScript management setup (eg, Sprockets, Webpack...), how I'm trying to use this other library, and why it's not working.
- [ ] I have another issue to discuss.

-------
I've followed the instructions in the readme to setup react-rails with webpacker and am trying to get a component, `HelloWorld.js` to render using `<%= react_component("HelloWorld") %>` in my application layout view. The helper properly converts to this div: `<div data-react-class="HelloWorld" data-react-props="{}"></div>` but the corresponding component is not rendered. 

I have a feeling this issue is related to my other [issue here](https://github.com/reactjs/react-rails/issues/721).

My project is available on [github here](https://github.com/chrismv48/debate_platform/tree/add-webpack-react).

**Comment by rmosolgo** [comment]
After making a pack with webpacker, you must include it with a `<script>` tag:

![image](https://cloud.githubusercontent.com/assets/2231765/26276488/7626403e-3d46-11e7-86d6-e9e11a07996f.png)


Since the pack wasn't included, `ReactRailsUJS` wasn't on the page. `ReactRailsUJS` is required to mount components. 

After adding that line, it worked for me: 

![image](https://cloud.githubusercontent.com/assets/2231765/26276497/95e6274a-3d46-11e7-9c01-6f3656683cf5.png)


Please reopen if that doesn't work for you!

**Comment by sksea** [comment]
Should this be done automatically by react install generator? I was confused when my components in `./app/javascripts/components` suddenly stopped rendering after running `rails g react:install` (after installing webpacker). Between docs for react-rails and webpacker, it wasn't obvious what was happening. Maybe I missed something.

Not sure if related, but this error appears in console.
![image](https://cloud.githubusercontent.com/assets/11454733/26279499/dc57591e-3d83-11e7-8fac-13155c3dff02.png)

**Comment by dsuare1** [comment]
I'm quite new to implementing React into a Rails app (been working on it for 1 day so far, lol) but what I notice is that when providing `{ prerender: true }`, the component does render on the page, but I can't seem to get an `onChange` event to fire in an input element. Is this because the component has been server-side rendered and thus the magic client-side rendering of React is mitigated? Any help is greatly appreciated!

For more context, I've included:

- `<%= javascript_pack_tag 'application' %>` in my `application.html.erb` layout file
- `<%= react_component("ComponentName", { props: @props }) %>` in my view file

**Comment by BookOfGreg** [comment]
@dsuare1 You may want to ask on Stack Overflow for these kinds of help questions, or open a new issue as it's quite easy for comments to get lost in closed issues.

For now though let's take a look. Server-side rendering doesn't affect client-side rendering as the components are rendered client-side anyway so there should not be a problem with that.
1) Are there any issues in the console?
2) Does it work if you do not prerender?
3) Are there any other JS libraries that may affect it such as Turbolinks or JQuery (Especially JQuery3 as there is a known bug with it)?

**Comment by dsuare1** [comment]
Thanks for your response @BookOfGreg; there are a few related questions on StackOverflow, but none of them helped me arrive at a solution.

1. There are no issues in the console that should be related to this problem.
2. It does not work if I don't prerender. Well, it's strange because when I don't pass `{ prerender: true }`, I can see the React Component inserted into the DOM, but whatever content is inside the Component does not render to the page, and is not present in the DOM. *See further explanation below
3. `Turbolinks` and `jquery-rails` as well `jquery-ui-rails` libraries are included, so that may be causing an issue

*To further explain Point 2 above, if I have a react component structured like:
`class HelloWorld extends React.Component {
  render() {
    return (
      <div>
        <h1>Hello, World</h1>
      </div>
    )
  }`
I can see that there is a `<div>` rendered to the page with `data-react-class=HelloWorld` and `data-react-props={}` attributes, but that inner `<h1>` content is not included as part of the <div> nor does it show up on the page.

**Comment by BookOfGreg** [comment]
Continuing on a hunch, can you run `ReactRailsUJS.mountComponents()` in your console? I suspect it's JQuery3 breaking the automatic mounting again.
If  so then it's just #762

**Comment by dsuare1** [comment]
@BookOfGreg Yes, calling `ReactRailsUJS.mountComponents()` manually in the console, once the page has loaded, does work and results in the component rendering. But I've read through 762 and the solution in there doesn't seem to fit my issue exactly. Deekor mentions moving the `//= require components` below the `//= require jquery3`, but I don't see a `//= require components` statement in my `app/assets/javascripts/application.js` file. Am I missing something in there?

**Comment by BookOfGreg** [comment]
It's a hacky solution for now but it is safe to do it that way. You will likely be using JQuery 3 and it broke it's support, so someone needs to update the file at https://github.com/reactjs/react-rails/blob/master/react_ujs/src/events/detect.js to play nicely with it.

**Comment by buck3000** [comment]
I also want to know the answer to this question...

**Comment by dsuare1** [comment]
@BookOfGreg Thanks for the help! You helped me get unblocked and keep my development moving forward. I'll keep an eye out for when / if this fix comes through, but for now, adding the manual call to `ReactRailsUJS.mountComponents` in `app/javascript/packs/application.js` works!

**Comment by kaylalarson1990** [comment]
Mine is still not working even when I paste `ReactRailsUJS.mountComponents()` at the end of my `app/javascript/packs/application.js` - has anyone else had any luck doing anything else?

**Comment by tOOnPT** [comment]
my hacky solution... in the view file where I call `<%= react_component(...` I just placed on the bottom:
```html
<script>
  $(document).ready(function() { ReactRailsUJS.mountComponents(); })
</script>
```
